docs/en/DevelopingExtensions.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: global
+group: Resources
+title: Developing Under Storage Extensions
+---
+
+* Table of Contents
```

Comment:
Could we make this a tutorial where we walk the user through building a toy UFS? This would be valuable for external developers trying to create new UFSes, and also valuable for us when we verify that the UFS connector story really works. We could put the toy example in `alluxio-extensions` for reference from here.